### ISSUE TITLE
NullReference exception in LegacyDataTypeChangingHandler.Handle

### ISSUE THREAD

**Issue by fcrocetti** [callumbwhyte/meganav#64]
The following code inside the second .Where predicate is giving a null reference exception when creating a new MegaNav data type because "existingEntity" is null when x.Id is equal to 0:

```
var dataTypes = notification.SavedEntities
                .Where(x => x.EditorAlias == Constants.PropertyEditorAlias)
                .Where(x =>
                {
                    var existingEntity = _dataTypeService.GetDataType(x.Id);

                    return existingEntity?.EditorAlias != Constants.PropertyEditorAlias && LegacyEditors.Aliases.Any(existingEntity.EditorAlias.InvariantContains);
                });
```

For now, to work around the issue I have added a null-check condition in the return predicate of LegacyEditors.Aliases.Any:

```
var dataTypes = notification.SavedEntities
                .Where(x => x.EditorAlias == Constants.PropertyEditorAlias)
                .Where(x =>
                {
                    var existingEntity = _dataTypeService.GetDataType(x.Id);

                    return existingEntity?.EditorAlias != Constants.PropertyEditorAlias && LegacyEditors.Aliases.Any((existingEntity?.EditorAlias ?? "").InvariantContains);
                });
```

**Comment by dawoe** [comment]
I can confirm this one.

**Comment by callumbwhyte** [comment]
Hey @fcrocetti and @dawoe,

Thanks for bringing this to my attention so quickly. Teaches me for only testing the release with the existing navs I had...

This is now fixed in v2.0.3 / v3.0.3 on NuGet, hopefully it didn't cause you too much inconvenience.

Cheers,
Callum
